openquake/hazardlib/gsim/atkinson_boore_2006.py

```diff
@@ -530,3 +530,25 @@ def _get_stress_drop_scaling_factor(self, magnitude):
         if stress_drop > cap:
             stress_drop = cap
         return log10(stress_drop / 140.0) / log10(2.0)
+
+
+class AtkinsonBoore2006SGS(AtkinsonBoore2006):
+    """
+    This class extends the original base class
+    :class:`openquake.hazardlib.gsim.atkinson_boore_2006.AtkinsonBoore2006`
+    by introducing a distance filter for the near field, as implemented
+    by SGS for the national PSHA model for Saudi Arabia.
+    """
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        Using a minimum distance of 5km for the calculation.
+        """
+
+        dists.rrup = np.array([d if d > 5. else 5. for d in dists.rrup])
```

Comment:
Perhaps it's faster this: 
dists.rrup[dists.rrup<=5] = 5.

Reply:
I tried your suggestion, but this has the drawback to modify also the original reference to the distance variable I'm using when calling the function, which is undesirable to me.
In oder words, the input distance array is returned modified to [5., 5., 5., 5., 10., 20., .....]
For reasons not fully clear to me, the first implementation doesn't do it.

Reply:
ok. I don't understand why it's a problem if dists.rrup is modified.

Reply:
@klunk386 I wrote this small example to check your assertion about the two methods and, if I understood correctly, the drawback occur also with your solution (luckily, because an object is mutable so you can change back its attributes values). To test it try `fun` and than `fun2` 
```python
#!/usr/bin/env python
import numpy as np


class Pippo():
    def __init__(self):
        self.rrup = None


def fun(pi):
    pi.rrup[pi.rrup <= 5] = 5


def fun2(pi):
    pi.rrup = np.array([d if d > 5. else 5. for d in pi.rrup])
    print("inner %s" % pi.rrup)


if __name__ == '__main__':
    pi = Pippo()
    pi.rrup = np.array([1., 2., 3., 4., 5., 6., 7., 8., 9.])
    print(pi.rrup)
    fun(pi)
    print(pi.rrup)
```